File: javascript/primality.js
/*
  Author: Vincent T. Mossman
  Updated:   July 25, 2018
*/

/*
isPrime returns TRUE if n is prime; else returns FALSE
*/
function isPrime(n) {

  // initialize divisor
  let div = 2;

  // determine lowest divisor (if any)
  while (n % div != 0) {
    if (div > Math.sqrt(n)) {
      div = n;
      break;
    } //end if
    div++;
  } //end while

  // determine primality
  if (div < n) {
    return false;
  } else {
    return true;
  } //end if

} //end isPrime

File: primality.c
/*
  Author: Vincent T. Mossman
  Date:   October 18, 2016
*/

#include <stdbool.h>

/******************************************************************************
* Function isPrime returns true if n is prime, else returns false             *
******************************************************************************/
bool isPrime(long int n) {
  
  long int i = 2;

  // find divisor (if any)
  while (n % i != 0) {
    if (i > (n / 2)) {
        i = n;
        break;
    }
    i++;
  } // end while

  // determine primality
  if (i < n) {
    return false;
  }
  else {
    return true;
  } // end if

} // end isPrime
